src/QATools/Example/Pages/LoginPage.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * This file is part of the QA-Tools library.
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @copyright Michael Iwersen <mi.iwersen@gmail.com>
+ * @link      https://github.com/qa-tools/behat-example
+ */
+
+namespace QATools\Example\Pages;
+
+
+use QATools\QATools\HtmlElements\Element\Form;
+use QATools\QATools\HtmlElements\TypifiedPage;
+
+/**
+ * Class LoginPage.
+ *
+ * @page-url('/')
+ */
+class LoginPage extends TypifiedPage
+{
+
+	/**
+	 * The login form.
+	 *
+	 * @var Form
+	 * @find-by('id' => 'login-form')
+	 */
+	protected $loginForm;
+
+	/**
+	 * Fill the login form with given data.
+	 *
+	 * @param string $username The username.
+	 * @param string $password The password.
+	 *
+	 * @return static
+	 */
+	public function fillForm($username, $password)
+	{
+		$this->loginForm->fill(array(
+			'username' => $username,
+			'password' => $password,
+		));
+
+		return $this;
+	}
+
+	/**
+	 * Does the login.
+	 *
+	 * @return Page
+	 */
+	public function loginSuccess()
```

Comment:
> but too my pity I couldn't get my hands on the tests of my colleagues.

You don't share code?

> If the submit step gets combined with entering the form the step might be look like ... the user is logging in successful with username and password. This might be too long for a step and too complex too understand. To make it short it could also be ... the user is logging in successful, I think this is what the stakeholder would write.

This should be 2 steps of course. First says, that `I login with ...` and second one should validate login result.

So the final steps might look like:
```gherkin
Given the user logins with 'username' and 'password'
Then login error message should say 'invalid username or password'
```
I'm not sure what should in step name: `the stuff that we check` (that error message is present on a page) or `fact, that login is ok or fail` (and leave detection of fail/check state as internal code in step implementation).